sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -167,6 +169,7 @@ protected void initChannel(final Channel channel) throws Exception {
              */
             @Override
             public void onTimeout(AcquireTask task) {
+                RntbdClientChannelPool.this.pendingAcquisitions.remove(checkNotNull(task, "expected non-null task"));
```

Comment:
Will there be any race condition here? 
what if the acquiredTask just being removed from runTasksInPendingAcquisitionQueue?

And also currently AcquireTimeout.run and AcquireTimeout.timeout both try to remove the task from the queue, so keep one is enough I think

Reply:
What scenario needs this. 

Timeout is scheduled on the pool eventLoop. 
When fired, will evicit timedout one from the pendingAcquisitionQueue and then only calls the onTimeout(). 
There is no other called for this onTimeut().